### PR TITLE
feat: exposing JSON and YAML spec endpoints in `@scalar/fastify-api-reference`, mimicking `@fastify/swagger-ui`

### PR DESCRIPTION
**Pull request by danwithabox**
**Problem**
https://github.com/scalar/scalar/issues/3065

**Solution**
- OpenAPI spec exposed on endpoints, just like how `@fastify/swagger-ui` does it, e.g. by default:
  - `reference/json `
  - `reference/yaml`
- these endpoints are configurable
- added relevant test cases
- the download button in `@scalar/fastify-api-reference` now points to these URLs except when an URL is explicitly provided

**Notes**
Exposing the endpoints was trivial:
 - used `@scalar/openapi-parser` and its `@scalar/openapi-parser/plugins/fetch-urls` plugin to always have a spec no matter how it was provided (`spec.content`, `spec.url`, or `fastify.swagger()`) in both JSON and YAML formats
 - the endpoints are tested to make sure they are reliably there

However, making the download button work as I would find it intuitive - always pointing to the exposed endpoints, except when `spec.url` is passed in the config - brought up some stuff:

1. ❌ **ditched idea:** first, in `@scalar/fastify-api-reference`, I simplified `DownloadLink.vue` and made the relevant changes to not have client side generation of the download at all
    - `@scalar/fastify-api-reference` would always pass down `spec.url`, pointing to the configured JSON endpoint
    - 🎉 was super clean 
    - but broke one test - `"has the spec"` in `fastifyApiReference.test.ts`
      - it didn't like that when `spec.content` was passed, I rewrote it to `spec.url`
      - unsure if this would be acceptable conceptually
    - plus, I realized this is beyond the scope of what I wanted and won't touch it without input from the maintainers
      - because other packages would have to change and expose the endpoints too
      - I am guessing these ones, but maybe more:
        - `@scalar/api-reference-react`
        - `@scalar/express-api-reference`
2. ❌ **ditched compromise:** download button in `DownloadLink.vue` unaffected, just expose endpoints
    - I simply think this is annoying, I expect to be able to find the download URL from the button, when available, without scouring the documentation, just like I did when using `@fastify/swagger-ui`
    - and the button is already capable of this, so it felt like a waste to ignore it!
3. ✅ **chosen compromise:** pass `spec.url` too to `@scalar/fastify-api-reference`, alongside `spec.content`
    - the test `"has the spec"` mentioned in idea 1/ does not fail
    - uses the existing logic of `DownloadLink.vue`: if it sees an URL, it switches to the `<a href>` link mode instead of the client side generation mode
      - 🎉 the endpoint is now easily discoverable when using `@scalar/fastify-api-reference`!
      - 🎉 using `@scalar/api-reference` in other wrappers with other backends, like React with Express, etc, is unaffected
    - ⚠️ however, I had to add one line to `standalone.ts` in `@scalar/api-reference` to use `getSpecUrl()` even when `spec.content` is detected
      - this feels a bit unclean, having both content and URL may not be semantically desirable? 🤷‍♂️
      - but could not think of a way to avoid changing `standalone.ts`
      - and everything seems to work, tests pass, I did not find code relying on an assumption of only one of them existing

I didn't like 2/ at all, but 1/ felt out of scope for now, so I went with 3/ as a sensible compromise.

Please do chip in about this choice, I will change this if needed, but I would think eventually going for 1/ is the most desirable.
Always having an exposed endpoint by design, and pointing the download button to it in all cases would just be very clean, imo.

Feel free to reword the `openApiSpecEndpoint` config and the JSDoc of it in `fastifyApiReference.ts`. I thought about not making it configurable for simplicity, but I think it's better to grant that flexibility, right?

Also, this is my first PR for OSS, if I missed something do tell!
Enjoyed contributing, the repo was set up rather ergonomically :)

Cheers!

### CODE REVIEW

**Review by hanspagel — CHANGES_REQUESTED** [review]
I love approach 3! Thanks for the awesome explanation!

I think we can *not* pass the content and just use the URL in the fastify integration. Actually, I think having support for directly passing the content was a mistake. 👀 

And I'd prefer to have the endpoints to look like files:

* reference/openapi.json 
* reference/openapi.yaml

Wdyt?

File: packages/fastify-api-reference/src/fastifyApiReference.ts
Comment (hanspagel) [inline packages/fastify-api-reference/src/fastifyApiReference.ts:44]: and can we name this `openApiDocumentEndpoints` (my fav), `openApiDocuments` or `openApiFiles`?

File: packages/oas-utils/package.json
Comment (hanspagel) [inline packages/oas-utils/package.json:104]: Can we stick to the other version? I think it’d increase the bundle size otherwise because one dependency is pinned to the other version.

Comment (danwithabox) [reply packages/oas-utils/package.json]: I didn't look around enough to have an input on that, and it's not like this _needs_ to be 2.5.1, so 2.4.5 it is 👍

File: packages/fastify-api-reference/src/fastifyApiReference.ts
Comment (danwithabox) [reply packages/fastify-api-reference/src/fastifyApiReference.ts]: renamed to `openApiDocumentEndpoints` 👍

**Comment by danwithabox** [comment]
YAML version change and renames done.

Endpoints changed to:
- reference/openapi.json
- reference/openapi.yaml

Now passing URL to `@scalar/fastify-api-reference`, always:
 - change `standalone.ts` in `@scalar/api-reference` is reverted, no longer processing both `.url` and `.content`, just like before
 - relevant tests reworked, added a test to check whether the explicitly passed URL is respected and not getting replaced by the exposed endpoint URL

Misc:
- fixed some eslint warnings

I think that's all the requested changes! 🎉

**Comment by hanspagel** [comment]
Best PR ever!

For some reason I can’t check out the PR, I think there’s something wrong on my machine. Once you/me/someone is able to fix the merge conflict, we’ll get this in ASAP.

**Review by hanspagel — APPROVED** [review]

**Comment by danwithabox** [comment]
Somehow the merge conflicted a completely safe change to `pnpm-lock.yaml` on origin, never seen this happen 😵‍💫 
Resolved it, synced my PR to origin/main, good to go in theory 👍

**Comment by hanspagel** [comment]
No worries. Actually, I don’t know what’s going on exactly, but I just want this PR in and we can further improve it in a separate PR. 😁

**Comment by hanspagel** [comment]
It’s in! :) Should be released later today, let’s see.

Thanks for this great PR, huge help to improve the fastify integration! 🙏

**Comment by danwithabox** [comment]
Cheers! If it ever becomes a goal to abolish the client-generated downloads en bloc, or any issue opens around Vue / TS / Fastify, I'm open for hire 😁

Can't wait for the release, can finally fully ditch Swagger UI 🥳

**Pull request merged**
